pkg/infoschema/infoschema_v2.go

```diff
@@ -455,6 +455,28 @@ func (is *infoschemaV2) EvictTable(schema, tbl string) {
 	is.tableCache.Remove(tableCacheKey{itm.tableID, itm.schemaVersion})
 }
 
+type helper struct {
+	end           tableItem
+	schemaVersion int64
+	ok            bool
+	itm           tableItem
+}
+
+func (h *helper) iter(item tableItem) bool {
+	if item.dbName != h.end.dbName || item.tableName != h.end.tableName {
+		h.ok = false
+		return false
+	}
+	if item.schemaVersion <= h.schemaVersion {
```

Comment:
It might happen when we're using historial read, or set tidb_snapshot, we may ask for the old infoschema, and the latest schema version has bumped. The btree records all the history version.